crawl-ref/source/travel.cc

```diff
@@ -4777,3 +4779,22 @@ int travel_trail_index(const coord_def& gc)
     else
         return -1;
 }
+
+bool stairs_destination_is_excluded(const stair_info &si)
+{
+    level_pos dest = si.destination;
+    LevelInfo &dest_li = travel_cache.get_level_info(dest.id);
+
+    if (is_unknown_stair(si.position) || !is_excluded(dest.pos, dest_li.get_excludes()))
+        return false;
+
+    const exclude_set &excludes = dest_li.get_excludes();
```

Comment:
What is the purpose of this second check? You seem to want to consider only some exclusions as "counting" for purposes of whether the stair destination is excluded. I imagine you have a good reason, but I don't know what it is. I suggest adding a comment explaining it.

Reply:
I meant to hit "request changes" here since I'd be happy to merge it once that comment is addressed.

Reply:
The purpose is to ignore single tile exclusions directly on the stair destination, which indicate that travel in one direction is excluded, but not the other (the current direction of travel). There might be such an exclusion in addition to other multi-tile exclusions on the destination floor, however, hence the loop. I'll add a comment.